docs/index.bs

```diff
@@ -3154,7 +3150,25 @@ spec: storage; urlPrefix: https://storage.spec.whatwg.org/
                       1. If |client| is not null, |response|'s [=response/type=] is "`opaque`", and [=cross-origin resource policy check=] with |request|'s [=request/origin=], |client|, "", and |response|'s [=filtered response/internal response=] returns <b>blocked</b>, then return null.
                       1. Return |response|.
               1. Return null.
-      1. If |request| is a <a>non-subresource request</a>, |request| is a [=navigation request=], |registration|'s [=navigation preload enabled flag=] is set, |request|'s [=request/method=] is \`<code>GET</code>\`, |registration|'s [=active worker=]'s [=set of event types to handle=] [=set/contains=] <code>fetch</code>, and |registration|'s [=active worker=]'s [=all fetch listeners are empty flag=] is not set then:
+          1. Else if |source| is {{RouterSourceEnum/"race-network-and-fetch-handler"}}, and |request|'s [=request/method=] is \`<code>GET</code>\` then:
+
+              Note: In order to avoid duplicated fetches to |request|, the user agent may reuse |raceNetworkRequestResponse| as the result of {{FetchEvent}}'s [=FetchEvent/potential response=], or the fallback request.
```

Comment:
This seems like it's specifying a possibly significant and observable step, via a vague non-normative note. Am I understanding correctly?

It seems to me you need to have some place to store the  _raceNetworkRequestResponse_, modify all the call sites that set the potential response, instead.

Reply:
Yes, your understanding is correct. Let me try writing this behavior in a spec level.

Reply:
Updated a bit to capture the case when the user agent reuses the network response for the fallback.

The other part is more problematic.

> the user agent may reuse |raceNetworkRequestResponse| as the result of {{FetchEvent}}'s [=FetchEvent/potential response=]

Actually this sentense was wrong. We have a dedupe mechanism for the |request| level, that means if `fetch(e.request)` is called inside the fetch handler, the user agent reuse the response dispatched before. But the response of `fetch(e.request)` is not `[=FetchEvent/potential response=]`. Partial response basically means the returned value in `e.respondWith()`.

I'm not sure how I can write this behavior in the spec, any advice would be appreciated.

Reply:
It sounds like you'd need to add quite a bit of spec machinery here, to parallel the implementation machinery. Probably that would involve a PR to https://fetch.spec.whatwg.org/ to modify the behavior of `fetch()`, or maybe one of the underlying algorithms like main fetch or HTTP-network fetch. And some sort of side table tracking the current response, etc.

Basically, how did you implement this in Chromium? You'll need to write up equivalent spec text.